### PR TITLE
Apply spacing options to Using and Catch

### PR DESCRIPTION
**Pull request by BradBarnich**
Makes `catch` and `using` follow `SpaceAfterControlFlowStatementKeyword` option.

Makes `using` follow `SpaceWithinOtherParentheses` option

Before, catch was always formatted as `catch (` and using as `using (var`

Now it is possible to have `catch(` and `using ( var`

### CODE REVIEW

**Comment by Pilchie** [comment]
@BradBarnich Is there an issue tracking this here on github yet?

@basoundr and/or @heejaechang want to take a look?

**Comment by BradBarnich** [comment]
@Pilchie I couldn't find one

**Comment by Pilchie** [comment]
@BradBarnich Okay - I'm pretty sure we have one, but it may be either still on CodePlex, or still in our internal TFS database.  Must get those issues out here pronto!

File: src/Workspaces/CSharp/Portable/Formatting/Rules/SpacingFormattingRule.cs [hunk 1]
```diff
@@ -102,7 +103,7 @@ public override AdjustSpacesOperation GetAdjustSpacesOperation(SyntaxToken previ
             if (currentKind == SyntaxKind.CloseParenToken &&
                 (currentParentKind == SyntaxKind.IfStatement || currentParentKind == SyntaxKind.WhileStatement || currentParentKind == SyntaxKind.SwitchStatement ||
                 currentParentKind == SyntaxKind.ForStatement || currentParentKind == SyntaxKind.ForEachStatement || currentParentKind == SyntaxKind.DoStatement ||
-                previousParentKind == SyntaxKind.CatchDeclaration))
+                previousParentKind == SyntaxKind.CatchDeclaration || currentParentKind == SyntaxKind.UsingStatement))
```
Comment (heejaechang) [inline src/Workspaces/CSharp/Portable/Formatting/Rules/SpacingFormattingRule.cs]: can you group this with currentParentKind.

**Comment by heejaechang** [comment]
:+1:

File: src/Workspaces/CSharp/Portable/Formatting/Rules/SpacingFormattingRule.cs (hunk 1 above)
Comment (BradBarnich) [inline src/Workspaces/CSharp/Portable/Formatting/Rules/SpacingFormattingRule.cs]: Sure. Were you thinking simply reorder the last two, so all currentParentKinds are before the previousParentKind?

Or more like this with previousParentKind broken out to a new line
```
if (currentKind == SyntaxKind.CloseParenToken &&
    (currentParentKind == SyntaxKind.IfStatement || currentParentKind == SyntaxKind.WhileStatement ||  currentParentKind == SyntaxKind.SwitchStatement ||
    currentParentKind == SyntaxKind.ForStatement || currentParentKind == SyntaxKind.ForEachStatement || currentParentKind == SyntaxKind.DoStatement ||
    currentParentKind == SyntaxKind.UsingStatement ||
    previousParentKind == SyntaxKind.CatchDeclaration))
```

Comment (heejaechang) [inline src/Workspaces/CSharp/Portable/Formatting/Rules/SpacingFormattingRule.cs]: ya, simply just re-ordering.

Comment (BradBarnich) [inline src/Workspaces/CSharp/Portable/Formatting/Rules/SpacingFormattingRule.cs]: Done

**Comment by basoundr** [comment]
:+1:

**Pull request merged**